dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/Clients/GeminiTextGenerationClient.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI.Core;
+
+/// <summary>
+/// Represents a client for interacting with the text generation gemini model.
+/// </summary>
+internal sealed class GeminiTextGenerationClient
+{
+    private readonly IGeminiChatCompletionClient _chatCompletionClient;
+
+    /// <summary>
+    /// Represents a client for interacting with the text generation gemini model.
+    /// </summary>
+    /// <param name="chatCompletionClient">A Gemini chat completion client instance.</param>
+    internal GeminiTextGenerationClient(IGeminiChatCompletionClient chatCompletionClient)
```

Comment:
Taking into account this dependency, do we need text generation client at all? Is there any difference from Google AI API point of view, or it's just a matter of usage in SK? Because if there is no functional difference between text generation and chat completion in Google AI API, I think it should be fine to release only chat completion client.

In OpenAI it's different case, because ITextGenerationService implementation supports different set of OpenAI models - text generation models.